app/screens/MachineDetails.js

```diff
@@ -106,7 +106,7 @@ class MachineDetails extends Component {
         const pintipsUrl = opdb_id ?
             `http://pintips.net/opdb/${opdb_id}` :
             ``
-
+        const operatorHasEmail = location.operator_id ? this.props.operators.operators.find(operator => operator.id === location.operator_id).operator_has_email : {}
```

Comment:
`operator_has_email` evaluates to a boolean, right? I'm also concerned chaining the `.operator_has_email` to the find. If something goes wrong, it will cause the app to crash. 
```suggestion
        const operator = location.operator_id ? this.props.operators.operators.find(operator => operator.id === location.operator_id)
        const operatorHasEmail = operator && operator.operator_has_email ? operator.operator_has_email : false
```